File: tareas-entregables/primera/pathlength.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Diagnostics;
namespace _3_12
{
    class Program
    {
        static double Pathlength(ref int[] x, ref int[] y)
        {
            double acc=0;
            for (int i = 1; i < x.Length; i++)
            {
                acc = acc + Math.Sqrt(Math.Pow(x[i] - x[i - 1], 2) + Math.Pow(y[i] - y[i - 1], 2));
            }
            return acc;
        }
        static void Test_Pathlength(ref int[] x, ref int[] y)
        {
            if (Pathlength(ref x, ref y) == 7)
            {
                Console.WriteLine("El metodo regresa el valor esperado");
            }
            else
            {
                Console.WriteLine("El metodo regresa un valor erroneo");
            }
        }
        static void Main(string[] args)
        {
            int[] x_points = { 0, 4, 6 };
            int[] y_points = { 0, 3, 3 };

            Test_Pathlength(ref x_points, ref y_points);

        }
    }
}

File: tareas-entregables/primera/pathlength.py
def pathlength(x,y):
    acc=0
    for i in range(1,len(x)):
        acc+=(pow(x[i]-x[i-1],2)+pow(y[i]-y[i-1],2))**(0.5)
    return acc
def test_pathlength():
    x=[0,2,6,8]
    y=[0,0,0,0]
    distance = pathlength(x,y)
    expected_result=9
    if(distance==expected_result):
        print(f'Resultado correcto')
        print(f'Valor esperado: {expected_result}, valor regresado: {distance}')
    else:
        print(f'Valor incorrecto regresado: {distance}')

if __name__ == '__main__':
    test_pathlength()